crates/ruma-state-res/src/event_auth.rs

```diff
@@ -338,12 +278,15 @@ pub fn auth_check<E: Event>(
 
     // If type is m.room.power_levels
     if *incoming_event.event_type() == TimelineEventType::RoomPowerLevels {
+        let room_power_levels_event = RoomPowerLevelsEvent::new(incoming_event);
         return check_room_power_levels(
-            &incoming_event,
-            power_levels_event.as_ref(),
+            room_power_levels_event,
+            current_room_power_levels_event,
             room_version,
             sender_power_level,
-        );
+        )
+        .map(|_| true)
+        .map_err(Error::custom);
```

Comment:
As mentioned earlier
```suggestion
        .map_or_else(Error::custom, |_| true);
```